File: Front-End/Main/model/Booking.js
/**
 * author : Sudeera Madushan
 * date : 11/1/2023
 * project : Front-End
 */

export class Booking {
    constructor(packageId, packageCategoryId,areaList,hotelList,vehicle,noOfChildren,totalHeadCount,dates,packageValue,customerId) {
        this._name = name;
        this._packageCategoryId = packageCategoryId;
        this._areaList = areaList;
        this._hotelList = hotelList;
        this._vehicle =vehicle;
        this._noOfChildren =noOfChildren;
        this._totalHeadCount =totalHeadCount;
        this._dates=dates;
        this._packageValue=packageValue;
        this._customerId=customerId;
    }

    get customerId() {
        return this._customerId;
    }

    set customerId(value) {
        this._customerId = value;
    }

    get packageValue() {
        return this._packageValue;
    }

    set packageValue(value) {
        this._packageValue = value;
    }

    get dates() {
        return this._dates;
    }

    set dates(value) {
        this._dates = value;
    }

    get noOfChildren() {
        return this._noOfChildren;
    }

    set noOfChildren(value) {
        this._noOfChildren = value;
    }

    get totalHeadCount() {
        return this._totalHeadCount;
    }

    set totalHeadCount(value) {
        this._totalHeadCount = value;
    }

    get name() {
        return this._name;
    }
    set name(newName) {
        this._name = newName;
    }

    set packageCategoryId(value) {
        this._packageCategoryId = value;
    }
    get packageCategoryId() {
        return this._packageCategoryId;
    }

    get areaList() {
        return this._areaList;
    }

    set areaList(value) {
        this._areaList = value;
    }

    get hotelList() {
        return this._hotelList;
    }

    set hotelList(value) {
        this._hotelList = value;
    }


    get vehicle() {
        return this._vehicle;
    }

    set vehicle(value) {
        this._vehicle = value;
    }
}

File: MainTravelService/src/main/java/lk/ijse/travel/maintravelservice/entity/Booking.java
package lk.ijse.travel.maintravelservice.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lk.ijse.travel.maintravelservice.dto.AreaDTO;
import lk.ijse.travel.maintravelservice.dto.DatesDTO;
import lk.ijse.travel.maintravelservice.dto.VehicleDTO;
import lombok.*;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

/**
 * @Author : Sudeera Madushan
 * @Date : 10/14/2023
 * @Project : Next Travel Pvt. Ltd
 */
@ToString
@Getter
@Setter
@Entity
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "booking")
public class Booking extends SuperEntity{
    @Column(name = "start_date")
    private Date startDate;
    @Column(name = "end_date")
    private Date endDate;
    @Column(name = "count_of_days")
    private int countOfDays;
    @Column(name = "count_of_nights")
    private int countOfNights;
    @Column(name = "no_of_children")
    private int noOfChildren;
    @Column(name = "total_headcount")
    private int totalHeadCount;
    @Column(name = "need_guid")
    private boolean needGuide;
    @Column(name = "with_pets")
    private boolean withPets;
    @Column(name = "package_value")
    private BigDecimal packageValue;
    @Column(name = "paid_value")
    private BigDecimal paidValue;
    private String remarks;
    @Column(name = "guide_id")
    private String guideId;
    @Column(name = "vehicle_id")
    private String vehicleId;
    @Column(name = "customer_id")
    private String customerId;
    @Column(name = "package_category_id")
    private String packageCategoryId;
    @OneToMany
    private List<Area> areaList=new ArrayList<>();
    @OneToMany
    private List<Dates> dates=new ArrayList<>();

}
